File: source/aoBlogs2/Models/BlogEntryModel.vb


Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses

Namespace Models
    ''' <summary>
    ''' Blog Posts
    ''' </summary>
    Public Class BlogEntryModel
        Inherits DbModel
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "Blog Entries"
        Public Const contentTableName As String = "ccBlogCopy"
        Private Shadows Const contentDataSource As String = "default"
        '
        '====================================================================================================
        ' -- instance properties
        '
        Public Property allowComments As Boolean
        Public Property anonymous As Boolean
        Public Property approved As Boolean
        Public Property articlePrimaryImagePositionId As Integer
        Public Property authorMemberID As Integer
        Public Property blogCategoryID As Integer
        Public Property blogID As Integer
        Public Property entryID As Integer
        Public Property imageDisplayTypeId As Integer
        Public Property primaryImagePositionId As Integer
        Public Property viewings As Integer
        Public Property podcastMediaLink As String
        Public Property podcastSize As Integer
        Public Property tagList As String
        Public Property copy As String
        '
        ''' <summary>
        ''' Return a list of Blog Copy
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <param name="blogId">The id of the Blog Copy</param>
        ''' <returns></returns>
        Public Shared Function createListFromBlogCopy(cp As CPBaseClass, blogId As Integer) As List(Of BlogEntryModel)
            Dim result As New List(Of BlogEntryModel)
            Try
                Dim Sql As String = "select p.ID" _
                        & " from (ccBlogCopy p" _
                        & " left join BlogCategories c on c.id=p.blogCategoryID)" _
                        & " where (p.blogid=" & blogId & ")" _
                        & " and((c.id is null)or(c.UserBlocking=0)or(c.UserBlocking is null))"
                result = createList(Of BlogEntryModel)(cp, "(id in (" & Sql & "))")
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
            Return result
        End Function
        '
        ''' <summary>
        ''' Return a list of Archive Blog Copy
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <param name="blogId">The id of the Blog Copy</param>
        ''' <returns></returns>
        Public Shared Function createArchiveListFromBlogCopy(cp As CPBaseClass, blogId As Integer) As List(Of ArchiveDateModel)
            Dim result As New List(Of ArchiveDateModel)
            Try
                'result = createList(cp, "(BlogID=" & blogId & ")", "year(dateadded) desc, Month(DateAdded) desc")
                Dim SQL = "SELECT DISTINCT month(dateadded) as archiveMonth, year(dateadded) as archiveYear" _
                            & " From ccBlogCopy" _
                            & " Where (ContentControlID = " & cp.Content.GetID(cnBlogEntries) & ") And (Active <> 0)" _
                            & " AND (BlogID=" & blogId & ")" _
                            & " ORDER BY year(dateadded) desc, month(dateadded) desc "
                Dim cs As CPCSBaseClass = cp.CSNew()
                If (cs.OpenSQL(SQL)) Then
                    Do
                        Dim archiveDate As New ArchiveDateModel()
                        archiveDate.Month = cs.GetInteger("archiveMonth")
                        archiveDate.Year = cs.GetInteger("archiveYear")
                        result.Add(archiveDate)
                        cs.GoNext()
                    Loop While cs.OK()
                End If

                cs.Close()
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
            Return result
        End Function

        Public Class ArchiveDateModel
            Public Year As Integer
            Public Month As Integer
        End Class

    End Class
End Namespace

File: source/aoBlogs2/Models/Db/BlogEntryModel.cs

using Contensive.BaseClasses;
using Contensive.Models.Db;
using System;
using System.Collections.Generic;

namespace Contensive.Blog.Models {
    public class BlogEntryModel : Contensive.Models.Db.DbBaseModel { 
        //
        // ====================================================================================================
        public static DbBaseTableMetadataModel tableMetadata { get; private set; } = new DbBaseTableMetadataModel("Blog Entries", "ccBlogCopy", "default", false);
        // -- const 
        // ====================================================================================================
        // -- instance properties
        /// <summary>
        /// if true, comments appear on this article
        /// </summary>
        /// <returns></returns>
        public bool allowComments { get; set; } = false;
        /// <summary>
        /// Determines how the primary image is displayed 1 = per stylesheet, 2 = right, 3 = left, 4 = hide
        /// </summary>
        public int primaryImagePositionId { get; set; }
        /// <summary>
        /// The user who last published this article.
        /// </summary>
        /// <returns></returns>
        public int authorMemberId { get; set; }
        /// <summary>
        /// Use for categorization. Like Tags but restricted
        /// </summary>
        /// <returns></returns>
        public int blogCategoryId { get; set; }
        /// <summary>
        /// The article text
        /// </summary>
        /// <returns></returns>
        public string copy { get; set; }
        /// <summary>
        /// comma delimited tag list
        /// </summary>
        /// <returns></returns>
        public string tagList { get; set; }
        /// <summary>
        /// number of times this article page is viewed
        /// </summary>
        /// <returns></returns>
        public int viewings { get; set; }
        /// <summary>
        /// Meta data for the page
        /// </summary>
        /// <returns></returns>
        public string metaTitle { get; set; }
        public string metaDescription { get; set; }
        public string metaKeywordList { get; set; }
        /// <summary>
        /// RSS data for the RSS feed
        /// </summary>
        /// <returns></returns>
        public DateTime rssDateExpire { get; set; }
        public DateTime rssDatePublish { get; set; }
        public string rssDescription { get; set; }
        public string rssLink { get; set; }
        public string rssTitle { get; set; }
        /// <summary>
        /// Add a video or audio file to the article. The RSS feed will create it as a podcast enclosure.
        /// </summary>
        /// <returns></returns>
        public string podcastMediaLink { get; set; }
        public int podcastSize { get; set; }
        // 
        public int blogId { get; set; }
        /// <summary>
        /// if provided, this is the public date displayed. Else, the dateAdded is used
        /// </summary>
        /// <returns></returns>
        public DateTime? datePublished { get; set; }
        public int blogpostpageid { get; set; }
        //
        // ==================================================================================================== 
        //
        public static void verifyPost(CPBaseClass cp, BlogEntryModel post) {
            if (post is null) { return; }
            if (post.dateAdded == null) {
                post.dateAdded = DateTime.Now;
                post.save(cp);
            }
            if (post.datePublished == null) {
                post.datePublished = post.dateAdded;
                post.save(cp);
            }
        }
        //
        // ==================================================================================================== 
        /// <summary>
        /// Return a list of Archive Blog Copy
        /// </summary>
        /// <param name="cp"></param>
        /// <param name="blogId">The id of the Blog Copy</param>
        /// <returns></returns>
        public static List<ArchiveDateModel> createArchiveListFromBlogCopy(CPBaseClass cp, int blogId) {
            var result = new List<ArchiveDateModel>();
            try {
                // result = createList(cp, "(BlogID=" & blogId & ")", "year(dateAdded) desc, Month(dateAdded) desc")
                string SQL = "SELECT DISTINCT month(dateAdded) as archiveMonth, year(dateAdded) as archiveYear" + " From ccBlogCopy" + " Where (ContentControlID = " + cp.Content.GetID(constants.cnBlogEntries) + ") And (Active <> 0)" + " AND (BlogID=" + blogId + ")" + " ORDER BY year(dateAdded) desc, month(dateAdded) desc ";



                var cs = cp.CSNew();
                if (cs.OpenSQL(SQL)) {
                    do {
                        var archiveDate = new ArchiveDateModel();
                        archiveDate.Month = cs.GetInteger("archiveMonth");
                        archiveDate.Year = cs.GetInteger("archiveYear");
                        result.Add(archiveDate);
                        cs.GoNext();
                    }
                    while (cs.OK());
                }

                cs.Close();
            } catch (Exception ex) {
                cp.Site.ErrorReport(ex);
            }
            return result;
        }
    }
}